src/plugins/vis_type_table_new/public/components/table_vis_component.tsx

```diff
@@ -0,0 +1,151 @@
+  title,
+  table,
+  visConfig,
+  event,
+  uiState,
+}: TableVisComponentProps) => {
+  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
+    table,
+    visConfig
+  );
+
+  const pagination = usePagination(visConfig, rows.length);
+
+  const sortedRows = useMemo(() => {
+    return uiState.sort && uiState.sort.colIndex !== null && uiState.sort.direction
+      ? orderBy(rows, columns[uiState.sort.colIndex]?.id, uiState.sort.direction)
+      : rows;
+  }, [columns, rows, uiState]);
+
+  const renderCellValue = useMemo(() => {
+    return (({ rowIndex, columnId }) => {
+      const rawContent = sortedRows[rowIndex][columnId];
+      const colIndex = columns.findIndex((col) => col.id === columnId);
+      const column = columns[colIndex];
+      // use formatter to format raw content
+      // this can format date and percentage data
+      const formattedContent = column.formatter.convert(rawContent, 'text');
+      return sortedRows.hasOwnProperty(rowIndex) ? formattedContent || null : null;
+    }) as EuiDataGridProps['renderCellValue'];
+  }, [sortedRows, columns]);
+
+  const dataGridColumns = getDataGridColumns(sortedRows, columns, table, event, uiState.width);
+
+  const sortedColumns = useMemo(() => {
+    return uiState.sort && uiState.sort.colIndex !== null && uiState.sort.direction
+      ? [{ id: dataGridColumns[uiState.sort.colIndex]?.id, direction: uiState.sort.direction }]
+      : [];
+  }, [dataGridColumns, uiState]);
+
+  const onSort = useCallback(
+    (sortingCols: EuiDataGridSorting['columns'] | []) => {
+      const nextSortValue = sortingCols[sortingCols.length - 1];
+      const nextSort: SortColumn =
+        sortingCols.length > 0
+          ? {
+              colIndex: dataGridColumns.findIndex((col) => col.id === nextSortValue?.id),
+              direction: nextSortValue.direction,
+            }
+          : {
+              colIndex: null,
+              direction: null,
+            };
+      uiState.setSort(nextSort);
+      return nextSort;
+    },
+    [dataGridColumns, uiState]
+  );
+
+  const onColumnResize: EuiDataGridProps['onColumnResize'] = useCallback(
+    ({ columnId, width }) => {
+      const curWidth: ColumnWidth[] = uiState.width;
+      const nextWidth = [...curWidth];
+      const nextColIndex = columns.findIndex((col) => col.id === columnId);
+      const curColIndex = curWidth.findIndex((col) => col.colIndex === nextColIndex);
+      const nextColWidth = { colIndex: nextColIndex, width };
+
+      // if updated column index is not found, then add it to nextWidth
+      // else reset it in nextWidth
+      if (curColIndex < 0) nextWidth.push(nextColWidth);
+      else nextWidth[curColIndex] = nextColWidth;
+
+      // update uiState.width
+      uiState.setWidth(nextWidth);
+    },
+    [columns, uiState]
+  );
+
+  const ariaLabel = title || visConfig.title || 'tableVis';
+
+  const footerCellValue = visConfig.showTotal
+    ? // @ts-expect-error
```

Comment:
why the typescript error?

Reply:
```
Binding element 'columnId' implicitly has an 'any' type.
```

Reply:
if you can't fix the typing it, an explicit cast is probably better than having the compiler ignore it altogether.

Reply:
okay I will bind the type